fe/fe-core/src/main/java/org/apache/doris/datasource/es/QueryBuilders.java

```diff
@@ -223,6 +239,20 @@ private static QueryBuilder parseFunctionCallExpr(Expr expr) {
         return new QueryBuilders.EsQueryBuilder(stringValue);
     }
 
+    private static String getColumnFromExpr(Expr expr) {
+        // Type transformed cast can not pushdown
+        if (expr instanceof CastExpr) {
+            Expr withoutCastExpr = exprWithoutCast(expr);
+            if (withoutCastExpr.getType().equals(expr.getType())
+                    || (withoutCastExpr.getType().isFloatingPointType() && expr.getType().isFloatingPointType())) {
+                return ((SlotRef) withoutCastExpr).getColumnName();
+            }
+        } else if (expr instanceof SlotRef) {
+            return ((SlotRef) expr).getColumnName();
+        }
+        return null;
```

Comment:
will cause npe in toEsDsl

Reply:
`column` will be checked in line 285.
And if `column` is `null`, `toESDsl` will return `null` and it will be handled by the caller.